lib/entity_linking.py

```
@@ -62,6 +64,43 @@
+
+def get_cache_key(text, service, **kwargs):
+    """Generate a unique fingerprint for a request."""
+    params = sorted(kwargs.items())
+    key_src = text.encode('utf-8') + service.encode('utf-8') + str(params).encode('utf-8')
+    return hashlib.md5(key_src).hexdigest()
+
+
+def get_cached_result(key):
+    """Retrieve result from disk cache."""
+    try:
+        db = shelve.open(CACHE_DB)
```

Ruff (0.15.7)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)